src/main/java/me/drepic/proton/common/Proton.java

```diff
@@ -0,0 +1,130 @@
+        this.config = getBootstrap().getConfiguration();
+        this.config.saveDefault();
+        this.config.loadConfig();
+
+        Logger logger = getBootstrap().getPluginLogger();
+
+        String clientName = this.config.getString("identification.clientName");
+        if (clientName == null) {
+            logger.log(Level.SEVERE, "The clientName must be set.");
+            getBootstrap().disable();
+            return;
+        }
+
+        String[] groups = config.getStringList("identification.groups").toArray(new String[0]);
+
+        if (!verifyIdentification(clientName, groups)) {
+            logger.log(Level.SEVERE, "The clientName/groups cannot contain `.` - Shutting down.");
+            getBootstrap().disable();
+            return;
+        }
+
+        boolean useRabbitMQ = config.getBoolean("rabbitMQ.useRabbitMQ");
+        boolean useRedis = config.getBoolean("redis.useRedis");
+
+        if (!useRabbitMQ && !useRedis) {
+            logger.log(Level.SEVERE, "Neither RabbitMQ nor Redis is enabled. Shutting down.");
+            getBootstrap().disable();
+            return;
+        }
+
+        if (useRabbitMQ) {
+            try {
+                setupRabbitMQ(clientName, groups);
+            } catch (IOException | TimeoutException e) {
+                e.printStackTrace();
+                logger.severe("Failed to setup RabbitMQ Connection");
+                getBootstrap().disable();
+                return;
+            }
+        } else {
+            setupRedis(clientName, groups);
+        }
+
+        boolean checkForUpdates = config.getBoolean("checkForUpdates");
+
+        if (checkForUpdates) {
+            try {
+                new UpdateChecker(this, getBootstrap().getVersion());
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    public final void disable(){
+        try {
+            ProtonManager manager = ProtonProvider.get();
+            manager.tearDown();
+            ProtonProvider.unregister();
+        }catch(Exception ignored){
+        }
+    }
+
+    private void setupRabbitMQ(String clientName, String[] groups) throws IOException, TimeoutException {
+        String host = this.config.getString("rabbitMQ.host");
+        String virtualHost = this.config.getString("rabbitMQ.virtualHost");
+        int port = this.config.getInt("rabbitMQ.port");
+        boolean useAuthorization = this.config.getBoolean("rabbitMQ.authorization.useAuthorization");
+
+        if (!useAuthorization) {
+            ProtonProvider.register(new RabbitMQManager(this, clientName, groups, host, virtualHost, port));
+        } else {
+            String username = this.config.getString("rabbitMQ.authorization.username");
+            String password = this.config.getString("rabbitMQ.authorization.password");
+            ProtonProvider.register(new RabbitMQManager(this, clientName, groups, host, virtualHost, port, username, password));
```

Comment:
you can have both setup methods return a manager and then you only need to call ProtonProvider.register() in the enable() method